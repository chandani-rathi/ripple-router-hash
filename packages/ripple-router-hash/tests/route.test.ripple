import { vi, describe, it, expect, beforeEach, afterEach } from 'vitest';
import { mount, flushSync} from 'ripple';
import { Route } from "@/route.ripple";

 async function waitFor(millisecond) {
	await new Promise((r) => setTimeout(() => r(), millisecond))
 }

describe('route', () => {

	let container;

	function render(component) {
		mount(component, {
			target: container
		});
	}

	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		document.body.removeChild(container);
		container = null;
	});

	it('render route with no props', async () => {
		console.log("compiled")
		component NoPropRoute() {
			<Route />
		}

		render(NoPropRoute);
        expect(container).toMatchSnapshot();

        flushSync(() => {});
		expect(container).toMatchSnapshot();

	});

    it('render route with empty $route', async () => {
		console.log("compiled")
		component EmptyRoute() {
			<Route $route={{}}/>
		}

		render(EmptyRoute);
        flushSync(() => {});
		expect(container).toMatchSnapshot();

	});

    
    it('render route with null component in $route', async () => {
		console.log("compiled")
		component NullComponentRoute() {
			<Route $route={{ path: "/", component: null}}/>
		}

		render(NullComponentRoute);
        flushSync(() => {});
		expect(container).toMatchSnapshot();

	});

     it('render route with empty path in $route', async () => {
		console.log("compiled");
        component Page(){
            <div>{"Hello Route"}</div>
        }

		component PageRoute() {
			<Route $route={{ path: "", component: () => Promise.resolve({Page})}}/>
		}

		render(PageRoute);
		flushSync(() => {})
		expect(container).toMatchSnapshot();

	});

     it('render route with $route', async () => {
		console.log("compiled");
        component Page(){
            <div>{"Hello Route"}</div>
        }

		const importModule = async () => {
			return { Page }
		}
		component PageRoute() {
			<Route $route={{ path: "/", component: importModule }}/>
		}

		render(PageRoute);
		flushSync(() => {});
		await waitFor(0)
		expect(container).toMatchSnapshot();

	});

     it('render route with inbuilt loading component $route', async () => {
		console.log("compiled");
        component Page(){
            <div>{"Hello Route"}</div>
        }

		const importModule = async () => {
			await waitFor(100);
			return { Page }
		}

		component PageRoute() {
			<Route $route={{ path: "/", component: importModule }}/>
		}

		render(PageRoute);
		flushSync(() => {});
		await new Promise((r) => setTimeout(() => r(), 0));
		expect(container).toMatchSnapshot();
		await waitFor(100);
		expect(container).toMatchSnapshot();
	});

     it('render route with loading component with loader $route', async () => {
		console.log("compiled");
        component Page({ data, params }){
            <div>{`Hello ${data.id}`}</div>
        }

        component Loading(){
            <div>{"Loading..."}</div>
        }

		async function loader({ params }){
			await waitFor(100);
			return { id: params.username }
		}

		const importModule = async () => {
			await waitFor(100);
			return { Loading, Page, loader }
		}

		component PageRoute() {
			<Route $route={{ path: "/", component: importModule, params: { username: "test"} }}/>
		}

		render(PageRoute);
		flushSync(() => {});
		await new Promise((r) => setTimeout(() => r(), 0));
		expect(container).toMatchSnapshot();
		await waitFor(100);
		expect(container).toMatchSnapshot();
		await waitFor(100);
		expect(container).toMatchSnapshot();
	});


    it('render route with inbuilt error component $route', async () => {
		console.log("compiled");
        component Page(){
            <div>{"Hello Route"}</div>
        }

		const importModule = async () => {
			await waitFor(100);
			throw Error("Test Error!")
			return { Page }
		}

		component PageRoute() {
			<Route $route={{ path: "/", component: importModule }}/>
		}

		render(PageRoute);
		flushSync(() => {});
		await new Promise((r) => setTimeout(() => r(), 0));
		await waitFor(100);
		expect(container).toMatchSnapshot();
	});
});