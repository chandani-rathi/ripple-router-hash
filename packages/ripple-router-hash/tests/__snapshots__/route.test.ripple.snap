// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`route > render route with $route 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div>
    Hello Route
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with empty $route 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <div
    class="min-h-screen bg-gradient-to-br from-gray-100 to-white p-8 flex flex-col items-center justify-center"
  >
    <h1
      class="text-4xl font-semibold"
    >
      404 - Not Found
    </h1>
    <div
      class="mt-4 text-gray-600"
    >
      The page you are looking for does not exist.
    </div>
    <a
      class="hover:underline"
      href="/"
    >
      Go to Home Page
      <!---->
    </a>
    <!---->
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with empty path in $route 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div
    class="min-h-screen bg-gradient-to-br from-gray-100 to-white p-8 flex flex-col items-center justify-center"
  >
    <h1
      class="text-4xl font-semibold"
    >
      404 - Not Found
    </h1>
    <div
      class="mt-4 text-gray-600"
    >
      The page you are looking for does not exist.
    </div>
    <a
      class="hover:underline"
      href="/"
    >
      Go to Home Page
      <!---->
    </a>
    <!---->
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with inbuilt error component $route 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div
    class="min-h-screen bg-gradient-to-br from-gray-100 to-white p-8 flex flex-col items-center justify-center"
  >
    <h1
      class="text-4xl font-semibold"
    >
      Error Occurred!
    </h1>
    <div
      class="mt-4 text-gray-600"
    >
      The page you are looking causing error.
    </div>
    <a
      class="hover:underline"
      href="/"
    >
      Go to Home Page
      <!---->
    </a>
    <!---->
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with inbuilt loading component $route 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div
    class="min-h-screen bg-gradient-to-br from-gray-100 to-white p-8 flex flex-col items-center justify-start"
  >
    Loading...
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with inbuilt loading component $route 2`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div>
    Hello Route
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with loading component with loader $route 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div
    class="min-h-screen bg-gradient-to-br from-gray-100 to-white p-8 flex flex-col items-center justify-start"
  >
    Loading...
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with loading component with loader $route 2`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div>
    Loading...
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with loading component with loader $route 3`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div>
    Hello test
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with loading state $route 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div
    class="min-h-screen bg-gradient-to-br from-gray-100 to-white p-8 flex flex-col items-center justify-start"
  >
    Loading...
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with loading state $route 2`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div>
    Hello Route
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with no props 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <div
    class="min-h-screen bg-gradient-to-br from-gray-100 to-white p-8 flex flex-col items-center justify-start"
  >
    Loading...
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with no props 2`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <div
    class="min-h-screen bg-gradient-to-br from-gray-100 to-white p-8 flex flex-col items-center justify-center"
  >
    <h1
      class="text-4xl font-semibold"
    >
      404 - Not Found
    </h1>
    <div
      class="mt-4 text-gray-600"
    >
      The page you are looking for does not exist.
    </div>
    <a
      class="hover:underline"
      href="/"
    >
      Go to Home Page
      <!---->
    </a>
    <!---->
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;

exports[`route > render route with null component in $route 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div
    class="min-h-screen bg-gradient-to-br from-gray-100 to-white p-8 flex flex-col items-center justify-center"
  >
    <h1
      class="text-4xl font-semibold"
    >
      404 - Not Found
    </h1>
    <div
      class="mt-4 text-gray-600"
    >
      The page you are looking for does not exist.
    </div>
    <a
      class="hover:underline"
      href="/"
    >
      Go to Home Page
      <!---->
    </a>
    <!---->
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;
